components/checkbox/index.css

```diff
@@ -158,7 +158,7 @@
 		}
 
 		&:hover {
-			.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+			.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box::before,
```

Comment:
I think the reason these styles weren't working is because we need the `.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box::before` selector.

As I was looking through the other hover styles, trying to understand why they were working but this one wasn't, it seems like every other hover instance is referencing the entire `.spectrum-Checkbox-input:checked + .spectrum-Checkbox-box::before` selector. Am I missing anything?  

I'm not sure I can correct this bug without it, but I'm very open to other ideas and happy to brainstorm!